app/controllers/homepage_controller.rb

```diff
@@ -243,8 +247,8 @@ def shapes
     ListingService::API::Api.shapes
   end
 
-  def location_search_params(latlng, search_engine_in_use)
-      if(feature_enabled?(:location_search) && latlng.present? && search_engine_in_use == :zappy)
+  def location_search_params(latlng, search_engine_in_use, main_search)
+      if(feature_enabled?(:location_search) && latlng.present? && search_engine_in_use == :zappy && main_search == :location)
```

Comment:
Maybe we can pass all search parameters to the adapter and let the search engine to decide which once to use?